Initial cleanup for 1.0.

portsmouth opened the pull request:
Add direct link to parameter reference from README.

portsmouth commented:
Clarified front page:

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/cd21eecd-5024-47a2-879f-0d12d9cb4330)

portsmouth commented:
The spec takes about 10 seconds to properly render, maybe we should mention that next to the white paper link?

portsmouth commented:
Added a copy-pastable BibTeX citation reference, which is needed if we want OpenPBR to be referenced in papers or documentation. (The form of this could be tweaked, we can check how it renders in a paper for example).

```
@techreport{OpenPBR,
  author = {Zap Andersson and Paul Edmondson and Julien Guertault and Adrien Herubel and Alan King and Peter Kutz and Andréa Machizaud and Jamie Portsmouth and Frédéric Servant},
  title = {OpenPBR Surface},
  institution = {Academy Software Foundation (ASWF)},
  year = {2024},
  url = {https://academysoftwarefoundation.github.io/OpenPBR/}
}
```

portsmouth commented:
Added a clickable GitHub icon to the spec (just after the version info), which links back to the GitHub page.

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/f275af1e-da3d-40fe-809a-9927b4ffac7d)

portsmouth commented:
The reference to the Gulbrandsen paper is broken.

portsmouth commented:
Rewording the text on the front page, as the sentence was long and hard to parse.

>OpenPBR Surface is a specification of a standard surface shading model for computer graphics. It aims to provide a material representation capable of accurately modeling the vast majority of CG materials used in practical visual effects and feature animation productions.

virtualzavie commented on README.md:
```diff
@@ -6,15 +6,17 @@
+### Specification
 
-* **[Specification](https://academysoftwarefoundation.github.io/OpenPBR/)**
+* **[White paper](https://academysoftwarefoundation.github.io/OpenPBR/)**
```
This line reverts the changes from #182. Is this intentional or an oversight?

portsmouth replied:
I think it's actually better like this, as under the "Specification" heading we have a number of components, i.e. the white paper, the parameter reference, the MaterialX graph, and the citation.

We say now "OpenPBR surface is a specification" (rather than e.g. "A white paper specifying" as for standard surface). But there is a white paper as part of that specification. Hopefully it makes sense..

virtualzavie replied:
Oh ok, I see the logic.

virtualzavie commented on README.md:
```diff
@@ -6,15 +6,17 @@
+* **[White paper](https://academysoftwarefoundation.github.io/OpenPBR/)**
+* **[Parameter reference](https://academysoftwarefoundation.github.io/OpenPBR/#parameterreference)**
 * **[Reference implementation](reference/open_pbr_surface.mtlx)** – written in [MaterialX](https://materialx.org/)
+* **[BibTeX citation](citation.bib)**
```
👍
Minor remark: I would have named it `openpbr.bib`, as a convenience for people downloading the file.

portsmouth replied:
Done in 5244f958157f855e51980e8d4098e5e87c51cfd3

virtualzavie commented on index.html:
```diff
@@ -2,22 +2,24 @@
 <meta name="viewport" content="width=device-width,initial-scale=1">
 
-**OpenPBR Surface specification**
+**OpenPBR Surface**
```
I question this change.
I think it's useful to see in the title that the document is a specification, especially considering that we may have more documents in the future, like an artist guide for example.

virtualzavie replied:
Note that "specification" and the "version" string could be moved to a sub-title before the author list.

portsmouth replied:
How about this? (Added in c81366e84ffbba7dfbdd5ca358b1640e51736657)

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/6595f510-ffd1-4d64-b1b8-257e067c3886)

virtualzavie replied:
Yep, I think that works.
(except of course for the additional changes mentioned in the meeting)

jstone-lucasfilm replied:
Should we include the removal of top-level company and author names as part of this changelist, as suggested in today's meeting?

portsmouth replied:
OK let me try it.

portsmouth replied:
>removal of top-level company 

Do we want to mention ASWF in its place?

virtualzavie commented on index.html:
```diff
@@ -2,22 +2,24 @@
 <center>Peter Kutz, Andréa Machizaud, Jamie Portsmouth, Frédéric Servant.</center>
 
-<center>*Version 0.4, 2024-04-30. A joint work of Adobe and Autodesk*</center>
+<center>*Version 0.4, 2024-04-30. A joint work of Adobe and Autodesk &nbsp; <a href="https://github.com/AcademySoftwareFoundation/OpenPBR" title="Redirect to GitHub page"><img src="images/github-mark.png" alt="homepage" width="24" style="vertical-align: middle; position: relative; bottom: 3px"/></a> *</center>
```
I see the value of having a link in the document, but I'm not a fan of a using a logo for that link.
I think a full text link would be clearer and look cleaner as well.

portsmouth replied:
Really? I find the GitHub logo fairly universal, and a neat way to indicate what the link is.

If we make it a text link, what is the text, and where.

virtualzavie replied:
Matter of taste I guess; that's a cosmetic detail anyway, let's move on to the rest.

virtualzavie commented on index.html:
```diff
@@ -1397,6 +1399,8 @@
 
 [#Gritz2010]: Larry Gritz, Clifford Stein, Chris Kulla, and Alejandro Conty. *Open Shading Language*, ACM SIGGRAPH Talks (2010).
 
+[#Gulbrandsen2014]: Ole Gulbrandsen. *Artist Friendly Metallic Fresnel*, Journal of Computer Graphics Techniques (2014).
```
👍

virtualzavie commented on citation.bib:
```diff
@@ -0,0 +1,7 @@
+  author = {Zap Andersson and Paul Edmondson and Julien Guertault and Adrien Herubel and Alan King and Peter Kutz and Andréa Machizaud and Jamie Portsmouth and Frédéric Servant},
+  title = {OpenPBR Surface},
+  institution = {Academy Software Foundation (ASWF)},
+  year = {2024},
```
I'd suggest to add `version` and `month` fields, as we may have subsequent versions published the same year.

virtualzavie commented:
> Someone with admin access is probably needed to do that, as it may require pointing the GitHub Pages link to the `internal/index.html` etc. See the dialog below for that.

Internal doesn't seem like a faithful description of the nature of those files. Accessing them directly is only problematic on GitHub, but works fine for someone who clones the repository.
What about `specification/index.html` or `spec/index.html`? This way they are moved away from top-level, but are not "hidden".

virtualzavie commented:
> The spec takes about 10 seconds to properly render (due to all the MathJax I assume), maybe we should mention that next to the white paper link?

On my side, it loads nearly instantaneously, both on Firefox/macOS, Firefox/Windows and Edge/Windows.

portsmouth commented:
> > The spec takes about 10 seconds to properly render (due to all the MathJax I assume), maybe we should mention that next to the white paper link?
> 
> On my side, it loads nearly instantaneously, both on Firefox/macOS, Firefox/Windows and Edge/Windows.

Do the equations render correctly nearly instantaneously?  On my system, they show up immediately, but change format as they are correctly rendered.

virtualzavie commented on openpbr.bib:
```diff
@@ -3,5 +3,7 @@ @techreport{OpenPBR
   institution = {Academy Software Foundation (ASWF)},
   year = {2024},
+  month = {May},
+  version = {1.0},
```
A note of caution to @jstone-lucasfilm for the release candidate, that we now have a third place where the version number appears.

portsmouth replied:
I'm not convinced a version number is needed in a citation. It can be assumed that the reference will want to refer to the latest version, which is available at the given link.

jstone-lucasfilm replied:
Agreed that the version number is not usually included in citations, so this may not end up being an issue in practice.

portsmouth replied:
I'll remove it for now.

portsmouth commented:
> > Someone with admin access is probably needed to do that, as it may require pointing the GitHub Pages link to the `internal/index.html` etc. See the dialog below for that.
> 
> Internal doesn't seem like a faithful description of the nature of those files. Accessing them directly is only problematic on GitHub, but works fine for someone who clones the repository. What about `specification/index.html` or `spec/index.html`? This way they are moved away from top-level, but are not "hidden".

Sounds fine to me. But I can't make this change myself as it will break the GitHub pages deployment. @jstone-lucasfilm would you mind to make this change?

jstone-lucasfilm commented:
@portsmouth For moving the specification to a new location, I'm happy to assist, but I'd recommend that we leave this as an improvement for after today's release candidate.

portsmouth commented:
> @portsmouth For moving the specification to a new location, I'm happy to assist, but I'd recommend that we leave this as an improvement for after today's release candidate.

Sure that's fine. How about we merge the other improvements in this PR, and leave that for a different PR?

jstone-lucasfilm commented:
@portsmouth Sure, I'm fine for this PR to be the first in a series of improvements over the next two weeks, but then maybe the title of the PR should be more specific?  :D

portsmouth commented:
> @portsmouth Sure, I'm fine for this PR to be the first in a series of improvements over the next two weeks, but then maybe the title of the PR should be more specific? :D

OK, changed the PR title.

How is this? The ASWF logo doubles as the link back to the OpenPBR GitHub, killing two birds with one stone.

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/10fedd6c-e296-47ab-b209-f99203508af3)


I moved the authors to the end, and took the liberty of moving @jstone-lucasfilm to that list, as that seems fairer to me considering your contribution. The BibTeX citation is updated accordingly, with the version removed.

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/39f6c0ad-21af-4376-b4a5-dcb0bcd235e4)

```
@techreport{OpenPBR,
  author = {Zap Andersson and Paul Edmondson and Julien Guertault and Adrien Herubel and Alan King and Peter Kutz and Andréa Machizaud and Jamie Portsmouth and Frédéric Servant and Jonathan Stone},
  title = {OpenPBR Surface},
  institution = {Academy Software Foundation (ASWF)},
  year = {2024},
  month = {May},
  url = {https://academysoftwarefoundation.github.io/OpenPBR/}
}
```

portsmouth commented:
Let me add the shader playground image as well...

virtualzavie commented:
I think we can thank a few more people who have made significant contributions in the last months.
Chris Kulla, Eugene d'Eon, Masuo Suzuki, Andrea Weidlich, André Mazzone and Stephen Hill come to mind.

virtualzavie commented on index.html:
```diff
@@ -1407,7 +1404,20 @@
+Peter Kutz,
+Andréa Machizaud,
+Jamie Portsmouth,
+Jonathan Stone
+and
+Frédéric Servant.
```
```suggestion
Frédéric Servant
and
Jonathan Stone.
```

virtualzavie commented on index.html:
```diff
@@ -2,22 +2,19 @@
 
-<center>Zap Andersson, Paul Edmondson, Julien Guertault, Adrien Herubel, Alan King, </center>
-<center>Peter Kutz, Andréa Machizaud, Jamie Portsmouth, Frédéric Servant.</center>
+<center>*Specification v0.4, 2024-04-30. &nbsp; &nbsp; <a href="https://github.com/AcademySoftwareFoundation/OpenPBR" title="Redirect to GitHub page"><img src="images/aswf-logo-black.svg" alt="homepage" height="18" style="vertical-align: middle; position: relative; bottom: 3px"/></a> *</center>
```
```suggestion
<center>*Specification v0.4, 2024-05-14. &nbsp; &nbsp; <a href="https://github.com/AcademySoftwareFoundation/OpenPBR" title="Redirect to GitHub page"><img src="images/aswf-logo-black.svg" alt="homepage" height="18" style="vertical-align: middle; position: relative; bottom: 3px"/></a> *</center>
```

virtualzavie replied:
Ok that change was not actually useful, since the version hasn't been bumped yet. Sorry about that. 🙃

portsmouth commented:
How's this?

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/32bc73e3-329e-4b99-bc13-f2c05b9095fb)

portsmouth commented:
> I think we can thank a few more people who have made significant contributions in the last months. Chris Kulla, Eugene d'Eon, Masuo Suzuki, Andrea Weidlich, André Mazzone and Stephen Hill come to mind.

Good point, thanks for the list. I updated it to:

![image](https://github.com/AcademySoftwareFoundation/OpenPBR/assets/2774364/998dd060-ce6d-4768-910a-0a5e9bc93893)

jstone-lucasfilm reviewed (approved):
This looks good to me, thanks @portsmouth.
